controllers/consoleplugin/consoleplugin_objects.go

```diff
@@ -162,94 +163,49 @@ func (b *builder) deployment(cmDigest string) *appsv1.Deployment {
-		"-frontend-config", filepath.Join(configPath, configFile),
-	}
-
-	if b.loki.TLS.Enable {
-		if b.loki.TLS.InsecureSkipVerify {
-			args = append(args, "-loki-skip-tls")
-		} else {
-			caPath := b.volumes.AddCACertificate(&b.loki.TLS, "loki-certs")
-			if caPath != "" {
-				args = append(args, "-loki-ca-path", caPath)
-			}
-		}
-	}
-
-	if b.loki.QuerierURL != b.loki.StatusURL {
-		args = append(args, "-loki-status", b.loki.StatusURL)
-		if b.loki.StatusTLS.Enable {
-			if b.loki.StatusTLS.InsecureSkipVerify {
-				args = append(args, "-loki-status-skip-tls")
-			} else {
-				statusCaPath, userCertPath, userKeyPath := b.volumes.AddMutualTLSCertificates(&b.loki.StatusTLS, "loki-status-certs")
-				if statusCaPath != "" {
-					args = append(args, "-loki-status-ca-path", statusCaPath)
-				}
-				if userCertPath != "" && userKeyPath != "" {
-					args = append(args, "-loki-status-user-cert-path", userCertPath)
-					args = append(args, "-loki-status-user-key-path", userKeyPath)
-				}
-			}
-		}
-	}
-
-	if b.loki.UseForwardToken() {
-		args = append(args, "-loki-forward-user-token")
-	} else if b.loki.UseHostToken() {
-		tokenPath := b.volumes.AddToken(constants.PluginName)
-		args = append(args, "-loki-token-path", tokenPath)
-	}
-
-	return args
-}
-
 func (b *builder) podTemplate(cmDigest string) *corev1.PodTemplateSpec {
-	volumes := []corev1.Volume{{
-		Name: secretName,
-		VolumeSource: corev1.VolumeSource{
-			Secret: &corev1.SecretVolumeSource{
-				SecretName: secretName,
+	volumes := []corev1.Volume{
+		{
+			Name: secretName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: secretName,
+				},
 			},
-		},
-	}, {
-		Name: configVolume,
-		VolumeSource: corev1.VolumeSource{
-			ConfigMap: &corev1.ConfigMapVolumeSource{
-				LocalObjectReference: corev1.LocalObjectReference{
-					Name: configMapName,
+		}, {
+			Name: configVolume,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: configMapName,
+					},
 				},
 			},
 		},
-	},
 	}
 
-	volumeMounts := []corev1.VolumeMount{{
-		Name:      secretName,
-		MountPath: "/var/serving-cert",
-		ReadOnly:  true,
-	}, {
-		Name:      configVolume,
-		MountPath: configPath,
-		ReadOnly:  true,
-	},
+	volumeMounts := []corev1.VolumeMount{
+		{
+			Name:      secretName,
+			MountPath: "/var/serving-cert",
+			ReadOnly:  true,
+		}, {
+			Name:      configVolume,
+			MountPath: configPath,
+			ReadOnly:  true,
+		},
 	}
 
-	args := b.buildArgs()
+	// ensure volumes are up to date
+	if b.loki.TLS.Enable && !b.loki.TLS.InsecureSkipVerify {
+		b.volumes.AddCACertificate(&b.loki.TLS, "loki-certs")
+	}
+	if b.loki.StatusTLS.Enable && !b.loki.StatusTLS.InsecureSkipVerify {
+		b.volumes.AddMutualTLSCertificates(&b.loki.StatusTLS, "loki-status-certs")
+	}
+	if b.loki.UseHostToken() {
+		b.volumes.AddToken(constants.PluginName)
+	}
```

Comment:
it seems like the volumes are added twice: here and from `setLokiConfig`. Is there a reason? Ideally if we could keep them from `setLokiConfig` it would be better, so we only have in a single place code related to checking TLS config

Reply:
All of these call the `insertOrReplace` function behind the scene
https://github.com/netobserv/network-observability-operator/blob/5fa07177c2ad1c3c9c39d88fdeb9eee40bfb1f35/pkg/volumes/builder.go#L133-L144

From what I seen, without that piece, the volumes are not up to date

Reply:
hmm, this is weird ... I'd prefer to keep the builder process as straightforward as possible, we should need this logic IMO .. I can take a look at that outside of your PR